lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Utils.pm

```diff
@@ -314,14 +314,17 @@ sub serialize_tags {
     if ($inc->user_tags) {
         $into->{'user-tags'} = [
             sort { $a->{name} cmp $b->{name} }
-            map +{ name => $_->tag->name }, @{ $opts->{user_tags} }
+            map +{ name => $_->tag->name },
+            grep { $_->is_upvote }
+                @{ $opts->{user_tags} }
         ];
     }
 
     if ($inc->user_genres) {
         $into->{'user-genres'} = [
             sort { $a->{name} cmp $b->{name} }
             map +{ disambiguation => $_->tag->genre->comment, id => $_->tag->genre->gid, name => $_->tag->name },
+            grep { $_->is_upvote }
                 @{ $opts->{user_genres} }
```

Comment:
Coding style looks weird. Personally I would always indent. Not sure about line returns and `,` separator.
@mwiencek: any suggestion?

Reply:
I guess I would indent the `grep` since `@{ $opts->{user_genres} }` is indented, and it's a continuation of the `map`.  The comma is needed because there are two forms of `map`: https://perldoc.perl.org/functions/map and here it's using `map EXPR,LIST`.  Since it's returning a hash ref, it avoids double brackets in the other form like `map { { id => ... } } @list`.